pkg/ingester/label_names_and_values.go

```diff
@@ -5,14 +5,25 @@ package ingester
 import (
 	"context"
 
+	"github.com/grafana/dskit/concurrency"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/tsdb"
 	"github.com/prometheus/prometheus/tsdb/index"
 
 	"github.com/grafana/mimir/pkg/ingester/client"
 )
 
-const checkContextErrorSeriesCount = 1000 // series count interval in which context cancellation must be checked.
+const (
+	checkContextErrorSeriesCount = 1000 // series count interval in which context cancellation must be checked.
+
+	maxConcurrency = 100
```

Comment:
This seems like a lot since the previous value was effectively `1`. We limit tenant federation to a max parallelism of `16` right now.

Reply:
Truth is that cardinality computation will take place mostly in memory (ingester's head), so you are right, setting such a high limit doesn't make much sense here. I'll change it. 👍